packages/legend-data-cube/src/stores/core/__tests__/DataCubeQueryFilterSnapshotBuilder.repl-test.ts

```diff
@@ -0,0 +1,168 @@
+import { DataCubeQueryFilterOperation__Equal } from '../filter/DataCubeQueryFilterOperation__Equal.js';
+import { DataCubeQueryFilterOperation__EqualCaseInsensitive } from '../filter/DataCubeQueryFilterOperation__EqualCaseInsensitive.js';
+import { DataCubeQueryFilterOperation__EqualCaseInsensitiveColumn } from '../filter/DataCubeQueryFilterOperation__EqualCaseInsensitiveColumn.js';
+import { DataCubeQueryFilterOperation__EqualColumn } from '../filter/DataCubeQueryFilterOperation__EqualColumn.js';
+import { DataCubeQueryFilterOperation__GreaterThan } from '../filter/DataCubeQueryFilterOperation__GreaterThan.js';
+import { DataCubeQueryFilterOperation__GreaterThanColumn } from '../filter/DataCubeQueryFilterOperation__GreaterThanColumn.js';
+import { DataCubeQueryFilterOperation__GreaterThanOrEqual } from '../filter/DataCubeQueryFilterOperation__GreaterThanOrEqual.js';
+import { DataCubeQueryFilterOperation__GreaterThanOrEqualColumn } from '../filter/DataCubeQueryFilterOperation__GreaterThanOrEqualColumn.js';
+import { DataCubeQueryFilterOperation__IsNotNull } from '../filter/DataCubeQueryFilterOperation__IsNotNull.js';
+import { DataCubeQueryFilterOperation__IsNull } from '../filter/DataCubeQueryFilterOperation__IsNull.js';
+import { DataCubeQueryFilterOperation__LessThan } from '../filter/DataCubeQueryFilterOperation__LessThan.js';
+import { DataCubeQueryFilterOperation__LessThanColumn } from '../filter/DataCubeQueryFilterOperation__LessThanColumn.js';
+import { DataCubeQueryFilterOperation__LessThanOrEqual } from '../filter/DataCubeQueryFilterOperation__LessThanOrEqual.js';
+import { DataCubeQueryFilterOperation__LessThanOrEqualColumn } from '../filter/DataCubeQueryFilterOperation__LessThanOrEqualColumn.js';
+import { DataCubeQueryFilterOperation__NotContain } from '../filter/DataCubeQueryFilterOperation__NotContain.js';
+import { DataCubeQueryFilterOperation__NotEndWith } from '../filter/DataCubeQueryFilterOperation__NotEndWith.js';
+import { DataCubeQueryFilterOperation__NotEqual } from '../filter/DataCubeQueryFilterOperation__NotEqual.js';
+import { DataCubeQueryFilterOperation__NotEqualCaseInsensitive } from '../filter/DataCubeQueryFilterOperation__NotEqualCaseInsensitive.js';
+import { DataCubeQueryFilterOperation__NotEqualCaseInsensitiveColumn } from '../filter/DataCubeQueryFilterOperation__NotEqualCaseInsensitiveColumn.js';
+import { DataCubeQueryFilterOperation__NotEqualColumn } from '../filter/DataCubeQueryFilterOperation__NotEqualColumn.js';
+import { DataCubeQueryFilterOperation__NotStartWith } from '../filter/DataCubeQueryFilterOperation__NotStartWith.js';
+import { DataCubeQueryFilterOperation__StartWith } from '../filter/DataCubeQueryFilterOperation__StartWith.js';
+import { DataCubeQueryFilterOperation__StartWithCaseInsensitive } from '../filter/DataCubeQueryFilterOperation__StartWithCaseInsensitive.js';
+
+type FilterSnapshotAnalysisTestCase = [
+  string, // name
+  string, // query roundtrip
+];
+
+const filterOperations = [
```

Comment:
let's use the list from `DataCubeEngine`, that's an abstract class, but you can create a new `engine` called `TEST__DataCubeEngine` and extract the filterOperations from there. I thought about making the filterOperations `static` but let's go with this for now, eventually, we will need to implement our own test engine anyway.

Reply:
done